### DISCUSSION TITLE
how to enable yocto toolchain integration with meson?

### DISCUSSION THREAD

**Discussion by cenit** [microsoft/vcpkg#41646] [Q&A] [answered]
I am cross compiling, both from windows and from linux, to a custom arm64-based triplet for a device that is running yocto. Toolchain and triplet seem fine for most ports from both host systems, at least now that i am still in the vcpkg exploration phase and i am still using simple ports. 
Unfortunately I am already stuck since I cannot find any way to fully integrate the meson ports, which have the bad habit of looking for dependencies outside of the sysroot passed to cmake.
Yocto already provides meson.native and meson.cross files in the /opt/mytoolchain/.../usr/share/meson sysroot. How to pass them to meson inside vcpkg to fix its behaviour? It seems to me that meson on vcpkg is programmed to take another approach, which is reading env from cmake for consistency (which i can understand). In this case, if I understood correctly, it seems that meson is not looking to all cmake variables (e.g. i think it's not reading/converting to its own language the `set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)` i have in the cmake toolchain file).
Any idea?

**Comment by dg0yt** [comment]
- Compare the machine files (yocto vs. vcpkg) for relevant differences.
- Create simple test ports.
- Know the limitations of meson, cmake and vcpkg :smile:  
  (Are you aware that `vcpkg_cmake_configure` doesn't set `VCPKG_HOST_TRIPLET`? Adding it changes `vcpkg.cmake` behavior with regard to `CMAKE_PROGRAM_PATH`.)

**Reply by dg0yt** [reply]
FTR https://github.com/microsoft/vcpkg/pull/38658 has some tests with regard to meson, but it wasn't too motivating.

**Reply by cenit** [reply]
where is the vcpkg file? Isn't it created in realtime extracting info from cmake?
I have simple test ports, and they do not work inside vcpkg

**Comment by cenit** [comment]
@Neumann-A I see that in `scripts\cmake\vcpkg_configure_meson.cmake` and in `ports\vcpkg-tool-meson\vcpkg_configure_meson.cmake` there is a `VCPKG_MESON_CROSS_FILE` defined. Is this the right variable to pilot to inject an external `meson.cross` file?
How? Is there any example

**Reply by Neumann-A** [reply, marked as answer]
simply `set(VCPKG_MESON_CROSS_FILE "${CMAKE_CURRENT_LIST_DIR}/your-meson-cross-file")` somewhere before the call. I advise to use the triplet but for testing it is probably better to do it in the port you want to use the cross file?

I mean it is just a variable so it works as every variable in cmake script mode.
